hibernate-core/src/main/java/org/hibernate/metamodel/mapping/internal/ToOneAttributeMapping.java

```diff
@@ -1065,7 +1065,15 @@ class Mother {
 			}
 			return false;
 		}
-		return parentNavigablePath.isSuffix( bidirectionalAttributePath );
+
+		NavigablePath navigablePath = parentNavigablePath.withoutDotIdentifierSequence( bidirectionalAttributePath );
+		if ( navigablePath != null ) {
+			if ( navigablePath.getLocalName().equals( EntityIdentifierMapping.ROLE_LOCAL_NAME ) ) {
```

Comment:
Could this also be a foreign key role?

Reply:
I thought about it , probably not but not 100% sure

Reply:
do you thing it is safer to add
```
final String localName = navigablePath.getLocalName();
			if ( localName.equals( EntityIdentifierMapping.ROLE_LOCAL_NAME )
					|| localName.equals( ForeignKeyDescriptor.PART_NAME )
					|| localName.equals( ForeignKeyDescriptor.TARGET_PART_NAME ) ) {
				navigablePath = navigablePath.getParent();
			}
```
?